### PR TITLE
Feature/vertical lines classes

### PR DESCRIPTION
**Pull request by dkarnutsch**
**Issue Number**

#364

**Overview of PR**

Custom styles for vertical lines based on date and unit

### CODE REVIEW

File: src/lib/lines/VerticalLines.js [hunk 1]
```diff
@@ -53,12 +56,18 @@ export default class VerticalLines extends Component {
           Math.ceil((nextTime.valueOf() - time.valueOf()) * ratio) - lineWidth
         const leftPush = firstOfType ? -1 : 0
 
+        let classNamesForTime = []
+        if (verticalLineClassNamesForTime) {
+          classNamesForTime = [verticalLineClassNamesForTime(time.unix() * 1000, nextTime.unix() * 1000 - 1)];
```
Comment (mcMickJuice) [inline src/lib/lines/VerticalLines.js]: Can you explain this line really quick:

* why `* 1000`?
* why subtract 1 from the `nextTime`?

**Comment by mcMickJuice** [comment]
Could you tidy up these merge conflicts?

Also, do you think this feature satisfies this open issue? https://github.com/namespace-ee/react-calendar-timeline/issues/228

If so, it would be awesome if you could provide an example in the examples folder of this repo on how to implement ranges

**Comment by dkarnutsch** [comment]
I resolved the conflicts, provided a new example for both holidays and ranges, as I think this might satisfies #228. If we get #368 to merge, I think we can cover most of the plugin use cases as you have mentioned.

File: demo/app/styles.scss [hunk 2]
```diff
@@ -50,3 +50,7 @@ body {
 .holiday {
   background-color: rgba(250, 246, 225, 0.5);
 }
+
+.lunch {
+  background-color: rgba(255, 0, 0, 0.5) !important; /* we need to overwrite the weekend rule */
```
Comment (mcMickJuice) [inline demo/app/styles.scss]: I think that with this feature implemented, we can remove the weekend styling, both from this repo codebase and from the style.scss file.

**Comment by AdamFreemer** [comment]
Forgive me, I'm still new to react, but I'm currently using the calendar in an app successfully and was trying to check out the functionality of this branch's feature of the vertical timeline. I loaded this branch into my package.json but webpacker wouldn't compile (with multiple errors I can list if it would help). 

I tried getting the demo running locally but got stuck at:
```
ERROR in ./demo/index.js
Module build failed: SyntaxError: Unexpected token (9:18)

   7 |
   8 | const render = AppToRender => {
>  9 |   ReactDOM.render(<AppToRender />, document.getElementById('root'))
     |                   ^
  10 | };
  11 |
  12 | render(App);

 @ multi demo
webpack: Failed to compile.
```
after working through some dependency issues. Any help greatly appreciated.

**Comment by dkarnutsch** [comment]
Yes, I would most definitely need more information to resolve your problem.

**Comment by AdamFreemer** [comment]
@dkarnutsch ok cool. Just a general question then, should this branch compile and run? I'm getting the following webpack errors below. The calendar was rendering off the main master branch. If it should compile as is, I can dig in to debugging. 

Just to confirm this is the correct way to point to a branch (in my package.json):
```    
"react-calendar-timeline": "github:dkarnutsch/react-calendar-timeline#feature/verticalLinesClasses",
```

```
ERROR in ./node_modules/react-calendar-timeline/index.js
Module not found: Error: Can't resolve './lib/lib/Timeline' in '/my/path/Code/didit_ross/node_modules/react-calendar-timeline'
resolve './lib/lib/Timeline' in '/my/path/Code/didit_ross/node_modules/react-calendar-timeline'
  using description file: /my/path/Code/didit_ross/node_modules/react-calendar-timeline/package.json (relative path: .)
    Field 'browser' doesn't contain a valid alias configuration
  after using description file: /my/path/Code/didit_ross/node_modules/react-calendar-timeline/package.json (relative path: .)
    using description file: /my/path/Code/didit_ross/node_modules/react-calendar-timeline/package.json (relative path: ./lib/lib/Timeline)
      no extension
        Field 'browser' doesn't contain a valid alias configuration
        /my/path/Code/didit_ross/node_modules/react-calendar-timeline/lib/lib/Timeline doesn't exist
      .js
        Field 'browser' doesn't contain a valid alias configuration
        /my/path/Code/didit_ross/node_modules/react-calendar-timeline/lib/lib/Timeline.js doesn't exist
      .sass
        Field 'browser' doesn't contain a valid alias configuration
        /my/path/Code/didit_ross/node_modules/react-calendar-timeline/lib/lib/Timeline.sass doesn't exist
      .scss
        Field 'browser' doesn't contain a valid alias configuration
        /my/path/Code/didit_ross/node_modules/react-calendar-timeline/lib/lib/Timeline.scss doesn't exist
      .css
        Field 'browser' doesn't contain a valid alias configuration
        /my/path/Code/didit_ross/node_modules/react-calendar-timeline/lib/lib/Timeline.css doesn't exist
```

**Comment by mcMickJuice** [comment]
@AdamFreemer this isn't the appropriate place to discuss this, you can open an issue in this repo to discuss how to reference git branches for this repo.

The problems you are facing are not specific to this PR, they're based on how this repo builds and publishes its code.

If your ultimate goal is to use the features introduced on this branch, I hope to publish them within the hour.

**Comment by AdamFreemer** [comment]
@mcMickJuice my question was to determine if it is a build issue with this branch, or a general build issue with this repo as you indicated. So you're confirming this branch builds without issue? As I stated, if that's the case, the problem is on my end and I will not comment here further.

**Comment by AdamFreemer** [comment]
@mcMickJuice I think you added your final comment on publishing the branch after I read it, so my apologies. I'll wait until this is merged and published and then debug on my end. Can't tell you guys enough how awesome this library is, appreciate your work. Thanks!

**Pull request merged**